test/cat.jl

```diff
@@ -88,27 +88,18 @@ module TestCat
     dt[1] = zeros(Int, nrow(dt))
     vcat(dt, alt_dt)
 
-    # Don't fail on non-matching names
-    names!(alt_dt, [:A, :B, :C])
-    vcat(dt, alt_dt)
-
     dtr = vcat(dt4, dt4)
     @test size(dtr, 1) == 8
     @test names(dt4) == names(dtr)
     @test isequal(dtr, [dt4; dt4])
 
-    dtr = vcat(dt2, dt3)
```

Comment:
Can you describe the new, stricter rules in detail in the commit message then? In general the message should detail all changes made and in particular give the rationale for them (which cannot always be inferred by reading the code), as well as tricky parts.

The rules probably need discussing too. It can make sense to throw an error when columns are not the same in all data tables, but it's not unreasonable to fill them with null values when they are missing either.